public/components/trace_analytics/components/traces/__tests__/span_detail_table.test.tsx

```diff
@@ -38,18 +272,17 @@ describe('<SpanDetailTable /> spec', () => {
 
   it('renders the component with data', async () => {
     const setCurrentSpan = jest.fn();
-    httpClientMock.post = jest.fn(() =>
-      Promise.resolve((TEST_SPAN_RESPONSE as unknown) as HttpResponse)
-    );
-    let container = document.createElement('div');
+    httpClientMock.post.mockResolvedValue((TEST_SPAN_RESPONSE as unknown) as HttpResponse);
+    const container = document.createElement('div');
     await act(() => {
       ReactDOM.render(
-        <SpanDetailTable
+        <SpanDetailTableHierarchy
           http={httpClientMock}
           hiddenColumns={['traceId', 'traceGroup']}
           DSL={{}}
           openFlyout={(spanId: string) => setCurrentSpan(spanId)}
-          mode='data_prepper'
+          mode="data_prepper"
+          dataSourceMDSId="testDataSource"
```

Comment:
Post snapshot check, we should add some tests to see if expand/collapse is working or not as that is the main functionality of this view.

Reply:
Covered with added cypress testing as follows:

describe('Testing traces tree view',
it('Verifies tree view and table toggle functionality with expand/collapse logic',
it('Verifies tree view expand arrow functionality',
it('Verifies span flyout',
it('Handles toggling between full screen and regular modes',